src/img/img-xpm.c

```diff
@@ -0,0 +1,1968 @@
+	{ 5234, 238, 180, 180 },
+	{ 5245, 205, 155, 155 },
+	{ 5256, 139, 105, 105 },
+	{ 5267, 65, 105, 225 },
+	{ 5277, 72, 118, 255 },
+	{ 5288, 67, 110, 238 },
+	{ 5299, 58, 95, 205 },
+	{ 5310, 39, 64, 139 },
+	{ 5321, 139, 69, 19 },
+	{ 5333, 250, 128, 114 },
+	{ 5340, 255, 140, 105 },
+	{ 5348, 238, 130, 98 },
+	{ 5356, 205, 112, 84 },
+	{ 5364, 139, 76, 57 },
+	{ 5372, 244, 164, 96 },
+	{ 5383, 46, 139, 87 },
+	{ 5392, 84, 255, 159 },
+	{ 5402, 78, 238, 148 },
+	{ 5412, 67, 205, 128 },
+	{ 5422, 46, 139, 87 },
+	{ 5432, 255, 245, 238 },
+	{ 5441, 255, 245, 238 },
+	{ 5451, 238, 229, 222 },
+	{ 5461, 205, 197, 191 },
+	{ 5471, 139, 134, 130 },
+	{ 5481, 160, 82, 45 },
+	{ 5488, 255, 130, 71 },
+	{ 5496, 238, 121, 66 },
+	{ 5504, 205, 104, 57 },
+	{ 5512, 139, 71, 38 },
+	{ 5520, 192, 192, 192 },
+	{ 5527, 135, 206, 235 },
+	{ 5535, 135, 206, 255 },
+	{ 5544, 126, 192, 238 },
+	{ 5553, 108, 166, 205 },
+	{ 5562, 74, 112, 139 },
+	{ 5571, 106, 90, 205 },
+	{ 5581, 131, 111, 255 },
+	{ 5592, 122, 103, 238 },
+	{ 5603, 105, 89, 205 },
+	{ 5614, 71, 60, 139 },
+	{ 5625, 112, 128, 144 },
+	{ 5635, 198, 226, 255 },
+	{ 5646, 185, 211, 238 },
+	{ 5657, 159, 182, 205 },
+	{ 5668, 108, 123, 139 },
+	{ 5679, 112, 128, 144 },
+	{ 5689, 255, 250, 250 },
+	{ 5694, 255, 250, 250 },
+	{ 5700, 238, 233, 233 },
+	{ 5706, 205, 201, 201 },
+	{ 5712, 139, 137, 137 },
+	{ 5718, 0, 255, 127 },
+	{ 5730, 0, 255, 127 },
+	{ 5743, 0, 238, 118 },
+	{ 5756, 0, 205, 102 },
+	{ 5769, 0, 139, 69 },
+	{ 5782, 70, 130, 180 },
+	{ 5792, 99, 184, 255 },
+	{ 5803, 92, 172, 238 },
+	{ 5814, 79, 148, 205 },
+	{ 5825, 54, 100, 139 },
+	{ 5836, 210, 180, 140 },
+	{ 5840, 255, 165, 79 },
+	{ 5845, 238, 154, 73 },
+	{ 5850, 205, 133, 63 },
+	{ 5855, 139, 90, 43 },
+	{ 5860, 0, 128, 128 },
+	{ 5865, 216, 191, 216 },
+	{ 5873, 255, 225, 255 },
+	{ 5882, 238, 210, 238 },
+	{ 5891, 205, 181, 205 },
+	{ 5900, 139, 123, 139 },
+	{ 5909, 255, 99, 71 },
+	{ 5916, 255, 99, 71 },
+	{ 5924, 238, 92, 66 },
+	{ 5932, 205, 79, 57 },
+	{ 5940, 139, 54, 38 },
+	{ 5948, 64, 224, 208 },
+	{ 5958, 0, 245, 255 },
+	{ 5969, 0, 229, 238 },
+	{ 5980, 0, 197, 205 },
+	{ 5991, 0, 134, 139 },
+	{ 6002, 238, 130, 238 },
+	{ 6009, 208, 32, 144 },
+	{ 6019, 255, 62, 150 },
+	{ 6030, 238, 58, 140 },
+	{ 6041, 205, 50, 120 },
+	{ 6052, 139, 34, 82 },
+	{ 6063, 245, 222, 179 },
+	{ 6069, 255, 231, 186 },
+	{ 6076, 238, 216, 174 },
+	{ 6083, 205, 186, 150 },
+	{ 6090, 139, 126, 102 },
+	{ 6097, 255, 255, 255 },
+	{ 6103, 245, 245, 245 },
+	{ 6114, 255, 255, 0 },
+	{ 6121, 255, 255, 0 },
+	{ 6129, 238, 238, 0 },
+	{ 6137, 205, 205, 0 },
+	{ 6145, 139, 139, 0 },
+	{ 6153, 154, 205, 50 }
+};
+
+enum buf_op { op_header, op_cmap, op_body };
+
+struct xpm_color {
+	char *color_string;
+	uint16_t red;
+	uint16_t green;
+	uint16_t blue;
+	int transparent;
+};
+
+struct file_handle {
+	FILE *infile;
+	char *buffer;
+	uint32_t buffer_size;
+};
+
+/* The following 2 routines (parse_color, find_color) come from Tk, via the Win32
```

Comment:
I pretty much rewrote these two routines in #2193. I think it's enough to avoid the extra copyright/licensing hassle (I am not a lawyer though...)